### PR TITLE
Meal vouchers - Add tests - Utils

### PR DESCRIPTION
**Pull request by araratthehero**
## Description
Add tests for giftcard and mealvoucher util classes

## Checklist 
- [x] PR is labelled 
- [x] Code is unit tested
- [x] Changes are tested manually

COAND-693

### CODE REVIEW

File: ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt [hunk 1]
```diff
@@ -11,9 +11,9 @@ package com.adyen.checkout.ui.core.internal.util
...
-enum class ExpiryDateValidationResult {
-    VALID,
-    INVALID_TOO_FAR_IN_THE_FUTURE,
-    INVALID_TOO_OLD,
-    INVALID_EXPIRY_DATE
+sealed class ExpiryDateValidationResult {
+    class Valid : ExpiryDateValidationResult()
+    class InvalidTooFarInTheFuture : ExpiryDateValidationResult()
+    class InvalidTooOld : ExpiryDateValidationResult()
+    data class InvalidExpiryDate(val isDateFormatInvalid: Boolean) : ExpiryDateValidationResult()
```
Comment (OscarSpruit) [inline ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt:18]: What is `isDateFormatInvalid` used for?

Comment (araratthehero) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt:18]: Expiry date can be invalid due to it being empty, misformatted or partially formatted. We need to check this in `CardValidationUtils:105`.

Comment (araratthehero) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt:18]: What I could do is create a different enum value for the specific case when isDateFormatInvalid is true to keep it more consistent and not pass any arguments.

Comment (OscarSpruit) [reply ui-core/src/main/java/com/adyen/checkout/ui/core/internal/util/ExpiryDateValidationResult.kt:18]: I think that would be more clear

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**